modules/dnn/src/layers/convolution_layer.cpp

```diff
@@ -150,6 +153,18 @@ class BaseConvolutionLayerImpl : public ConvolutionLayer
         }
         fusedWeights = false;
         fusedBias = false;
+        if (inputs[0].dims == 3)
+        {
+            //Conv1D
+            #define REMOVE_EXTRA_ELEM(vector)\
+            if (vector.size() > 1)\
+                vector.pop_back();
+            REMOVE_EXTRA_ELEM(kernel_size);
```

Comment:
There is no such code on "master" branch.

Reply:
@sl-sergei Any feedback?

Reply:
> @sl-sergei Any feedback?

@alalek I updated it, now everything should be ok